openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+        Returns True on errors.
+        """
+        errors = False
+        for template_field in self.get_fields(collection):
+            if not isinstance(template_field, BaseTemplateField):
+                continue
+            field_error = False
+            replacements = model.get(template_field.get_structured_field_name(""))
+
+            if not isinstance(replacements, list):
+                self.errors.append(
+                    f"{collection}/{model['id']}/{template_field.own_field_name}: Replacements for the template field must be a list"
+                )
+                field_error = True
+                continue
+            for replacement in replacements:
+                if not isinstance(replacement, str):
+                    self.errors.append(
+                        f"{collection}/{model['id']}/{template_field.own_field_name}: Each replacement for the template field must be a string"
+                    )
+                    field_error = True
+            if field_error:
+                errors = True
+                continue
+            replacement_collection = None
+            if template_field.replacement_collection:
+                replacement_collection = (
+                    template_field.replacement_collection.collection
+                )
+
+            for replacement in replacements:
+                structured_field = self.make_structured(template_field, replacement)
+                if structured_field not in model:
+                    self.errors.append(
+                        f"{collection}/{model['id']}/{template_field.own_field_name}: Missing {structured_field} since it is given as a replacement"
+                    )
+                    errors = True
+
+                if replacement_collection:
+                    try:
+                        as_id = int(replacement)
+                    except (TypeError, ValueError):
+                        self.errors.append(
+                            f"{collection}/{model['id']}/{template_field.own_field_name}: Replacement {replacement} is not an integer"
+                        )
+                    if not self.find_model(replacement_collection, as_id):
+                        self.errors.append(
+                            f"{collection}/{model['id']}/{template_field.own_field_name}: Replacement {replacement} does not exist as a model of collection {replacement_collection}"
+                        )
+
+            for field in model.keys():
+                if self.is_structured_field(field):
+                    try:
+                        _template_field, _replacement = self.to_template_field(
+                            collection, field
+                        )
+                        if (
+                            template_field.own_field_name == _template_field
+                            and _replacement not in model[template_field.own_field_name]
```

Comment:
Here it is actually relevant: `model[template_field.own_field_name]` will never exist because of the missing `$`

Reply:
Use get_own_field_name here.